redis/src/main/java/org/apache/calcite/adapter/redis/RedisDataBase.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.adapter.redis;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Set;
+
+import redis.clients.jedis.Jedis;
+
+/**
+ * The abstract class with RedisDataBase.
+ */
+public abstract class RedisDataBase {
+  String tableName;
+  String dataFormat;
+  String keyDelimiter;
+  RedisDataType dataType = null;
+  RedisDataFormat redisDataFormat = null;
+  List<LinkedHashMap<String, Object>> fields;
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  void init(Jedis jedis, RedisTableFieldInfo tableFieldInfo) {
```

Comment:
Can this be a constructor instead of `init()` method ? Is there a reason not to use constructor ? 

What about abstract methods like `jedis()` / `tableFieldInfo()` provided by subclasses.